gapir/cc/memory_allocator.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GAPIR_MEMORY_ALLOCATOR_H
+#define GAPIR_MEMORY_ALLOCATOR_H
+
+#include <map>
+#include <memory>
+
+namespace gapir {
+
+class MemoryAllocator;
+
+class MemoryAllocator {
+ private:
+  class MemoryRegion {
+   public:
+    MemoryRegion() : _offset(0), _size(0) {}
+    MemoryRegion(size_t offset, size_t size) : _offset(offset), _size(size) {}
+    virtual ~MemoryRegion() {}
+
+    size_t getOffset() const { return _offset; }
+    size_t getSize() const { return _size; }
+
+    bool operator<(const MemoryRegion& rhs) const {
+      return _offset < rhs._offset;
+    }
+
+   private:
+    size_t _offset;
+    size_t _size;
+  };
+
+ public:
+  class Handle {
```

Comment:
Add a comment on what is the purpose of this Handle class.

Reply:
Added